File: python/0231.py
class Solution(object):
    def isPowerOfTwo(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n < 1:
            return False
        while n > 1:
            if n % 2 == 1:
                return False
            n /= 2
        return True

File: 0231.c
bool isPowerOfTwo(int n) {
    if (n < 1) {
        return false;
    }
    else if (n == 1) {
        return true;
    }
    
    while (n > 1) {
        if (n & 1 == 1) {
            return false;
        }
        n >>= 1;
    }
    return true;
}